gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/hive/HiveCopyEntityHelper.java

```diff
@@ -354,35 +314,24 @@
           this.existingTargetTable = Optional.absent();
         }
 
-        // Constructing CommitStep object for table registration
-        Path targetPath = getTargetLocation(this.dataset.fs, this.targetFs, this.dataset.table.getDataLocation(),
-            Optional.<Partition> absent());
+        Path targetPath = getTargetLocation(this.targetFs, this.dataset.table.getDataLocation(), Optional.<Partition>absent());
+        if (MapUtils.getBoolean(this.dataset.getProperties(), ConfigurationKeys.USE_DATASET_LOCAL_WORK_DIR, false)) {
+          this.dataset.datasetPath = targetPath.toUri().getRawPath();
```

Comment:
Why does datasetPath depend on whether or not we are using dataset local working dir?